repl/scala-2.12/src/main/scala/org/apache/spark/repl/SparkILoop.scala

```diff
@@ -37,7 +37,14 @@ class SparkILoop(in0: Option[BufferedReader], out: JPrintWriter)
     @transient val spark = if (org.apache.spark.repl.Main.sparkSession != null) {
         org.apache.spark.repl.Main.sparkSession
       } else {
-        org.apache.spark.repl.Main.createSparkSession()
+        try {
+          org.apache.spark.repl.Main.createSparkSession()
+        } catch {
+          case e: Exception =>
+            println("Failed to initialize Spark session:")
+            e.printStackTrace()
+            sys.exit(1)
+        }
```

Comment:
+1, LGTM for `spark-shell` and `pyspark`. For consistency, we are going to change `sparkR` and `spark-sql` later in this way, aren't we? Could you mention `spark-shell` and `pyspark` in the PR title specifically?

Reply:
I'm not that familiar with those two shells but I'll give it a try.

Reply:
Thank you, @vanzin .

Reply:
SparkR already does something close to this:

```
  # 0 is success and +1 is reserved for heartbeats. Other negative values indicate errors.
  if (returnStatus < 0) {
    stop(readString(conn))
  }
```

Except that "stop()" doesn't actually exit the shell. That's probably by design, so unless @felixcheung  says it's ok to exit the shell (and tells me how), I'll leave things as is.

Reply:
Also, spark-sql for me exits right away if the SparkContext does not come up, showing the underlying exception, so it looks good already.

Reply:
there's a way, but overall we do not exit/terminate the R session since SparkR could be running in an interactive session (eg. RStudio)

one possible approach is to exit only when running sparkR shell, by checking here:
https://github.com/apache/spark/blob/master/R/pkg/inst/profile/shell.R#L27

I'm not sure if stop() vs exit makes much of a difference though.